sys/net/gnrc/sock/gnrc_sock.c

```diff
@@ -266,8 +285,15 @@ ssize_t gnrc_sock_send(gnrc_pktsnip_t *payload, sock_ip_ep_t *local,
             else {
                 type = payload->type;
             }
-            hdr = pkt->data;
-            hdr->nh = nh;
+
+            gnrc_pktsnip_t *ext = _last_ext_hdr(pkt);
+            if (ext) {
+                ipv6_ext_t *ext_hdr = ext->data;
+                ext_hdr->nh = nh;
+            } else {
+                hdr = pkt->data;
+                hdr->nh = nh;
+            }
```

Comment:
```suggestion
            if (ext->type == GNRC_NETTYPE_IPV6_EXT ) {
                ipv6_ext_t *ext_hdr = ext->data;
                ext_hdr->nh = nh;
            } else {
                hdr = pkt->data;
                hdr->nh = nh;
            }
```

Reply:
like this

Reply:
Well if you insist…

Reply:
this also makes it clear that there is a check and then a cast.

also make it more obvious that there are different types ( the reason i missed that in the first place)